relayer/provider/cosmos/msg.go

```diff
@@ -31,6 +32,18 @@ func CosmosMsg(rm provider.RelayerMessage) sdk.Msg {
 	}
 }
 
+// TypedCosmosMsg does not accept nil. IBC Message must be of the requested type.
+func TypedCosmosMsg[T *chantypes.MsgRecvPacket | *chantypes.MsgAcknowledgement](msg provider.RelayerMessage) T  {
```

Comment:
Does this function need to be exported/public?

AFAICS, it's only used in this `cosmos` package. Maybe future plans to use it outside the package? (In that case, I'd still make it private and refactor to public if/when necessary.)

The doc comment is 💯

Reply:
Made private for now